hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerClusterState.java

```diff
@@ -342,13 +347,12 @@ public String getRack(ServerName server) {
     }
 
     for (int i = 0; i < regionsPerServer.length; i++) {
-      primariesOfRegionsPerServer[i] = new int[regionsPerServer[i].length];
+      primariesOfRegionsPerServer[i] = new Int2IntCounterMap(
+        regionsPerServer[i].length, Hashing.DEFAULT_LOAD_FACTOR, 0);
       for (int j = 0; j < regionsPerServer[i].length; j++) {
         int primaryIndex = regionIndexToPrimaryIndex[regionsPerServer[i][j]];
-        primariesOfRegionsPerServer[i][j] = primaryIndex;
+        primariesOfRegionsPerServer[i].getAndIncrement(primaryIndex);
```

Comment:
Better rename the primariesOfRegionsPerServer to a name with something like count or counter in it? I think in the past we record the primaryIndex for all the regions on this region server but now, we just record the number of region replicas for a primary region on this region server?